### PR TITLE
Fix/285

### PR DESCRIPTION
**Pull request by rayrutjes**
Resolves: #285

### CODE REVIEW

**Comment by JanPetr** [comment]
Wow, cannot believe we missed such an issue.
The fix introduces a BC break which will break already "broken" implementations. But I cannot see any "non-BC-break" way how to fix it.

**Comment by rayrutjes** [comment]
Technically I agree with you @JanPetr , but it is more of a bug fix in this case.
If people are relying on bugs, then we cannot fix them.

For this one, I think we are good, and for eventual breaking apps, the fix will be more than obvious.

File: tests/AlgoliaSearch/Tests/IndexTest.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+class IndexTest extends AlgoliaSearchTestCase
+{
+    /**
+     * @expectedException \BadMethodCallException
+     * @expectedExceptionMessage No method named unknown was found.
+     */
+    public function testShouldThrowAnExceptionIfUnknownMethodIsCalled()
+    {
+        $clientContextMock = $this->getMockBuilder('AlgoliaSearch\ClientContext')->disableOriginalConstructor()->getMock();
```
Comment (JanPetr) [inline tests/AlgoliaSearch/Tests/IndexTest.php]: What is the reason behind using mock instead of real Client(Context) with passed env API credentials? 
Would it make sense to do it as unit test in a different test suite?

Comment (rayrutjes) [inline tests/AlgoliaSearch/Tests/IndexTest.php]: Here I mock the dependencies to test the Index class in isolation.
This is a unit test.

Comment (JanPetr) [inline tests/AlgoliaSearch/Tests/IndexTest.php]: I know it's a unit test. I was wondering if it makes sense to write a unit test and if it does if it's not better to write it in a different namespace / folder to create separate (unit and integration) test suites.

Comment (rayrutjes) [inline tests/AlgoliaSearch/Tests/IndexTest.php]: Yes we could separate integration from unit tests.
Let's do it in another PR though because we would need to separate more things than only this test.
